poncho/src/poncho/library_network_code.py

```diff
@@ -123,8 +123,20 @@ def start_function(in_pipe_fd, thread_limit=1):
                         event = cloudpickle.load(f)
 
                     # output of execution should be dumped to outfile.
-                    with open('outfile', 'wb') as f:
-                        cloudpickle.dump(globals()[function_name](event), f)
+                    result = globals()[function_name](event)
+                    try:
+                        with open('outfile', 'wb') as f:
+                            cloudpickle.dump(result, f)
+                    except Exception:
+                        if os.path.exits('outfile'):
+                            os.remove('outfile')
+                        raise
+
+                    try:
+                        if not result["Success"]:
+                            raise Exception(result["Reason"])
+                    except Exception:
+                        raise
```

Comment:
Same as above.

Reply:
Exception is caught and child exits. This child exits anyway when call finishes, this just changes the exit call on failure.